src/compiler/commandLineParser.ts

```diff
@@ -710,13 +714,259 @@ namespace ts {
       * @param fileName The path to the config file
       * @param jsonText The text of the config file
       */
-    export function parseConfigFileTextToJson(fileName: string, jsonText: string, stripComments = true): { config?: any; error?: Diagnostic } {
+    export function parseConfigFileTextToJson(fileName: string, jsonText: string): { config?: any; error?: Diagnostic } {
+        const jsonSourceFile = parseJsonText(fileName, jsonText);
+        return {
+            config: convertToJson(jsonSourceFile, jsonSourceFile.parseDiagnostics),
+            error: jsonSourceFile.parseDiagnostics.length ? jsonSourceFile.parseDiagnostics[0] : undefined
+        };
+    }
+
+    /**
+      * Read tsconfig.json file
+      * @param fileName The path to the config file
+      */
+    export function readConfigFileToJsonSourceFile(fileName: string, readFile: (path: string) => string): JsonSourceFile {
+        let text = "";
         try {
-            const jsonTextToParse = stripComments ? removeComments(jsonText) : jsonText;
-            return { config: /\S/.test(jsonTextToParse) ? JSON.parse(jsonTextToParse) : {} };
+            text = readFile(fileName);
         }
         catch (e) {
-            return { error: createCompilerDiagnostic(Diagnostics.Failed_to_parse_file_0_Colon_1, fileName, e.message) };
+            return <JsonSourceFile>{ parseDiagnostics: [createCompilerDiagnostic(Diagnostics.Cannot_read_file_0_Colon_1, fileName, e.message)] };
+        }
+        return parseJsonText(fileName, text);
+    }
+
+    const tsconfigRootOptions: CommandLineOption[] = [
+        {
+            name: "compilerOptions",
+            type: "object",
+            optionDeclarations: optionDeclarations,

```

Comment:
But those are two different hierarchies isn't it? The options specified here can only be present at root of json object and other command line options are possible only inside object called compiler Options?